src/Rectification.jl

```diff
@@ -353,3 +411,131 @@ function isbounded(r::Rectification{N})::Bool where {N<:Real}
+
+Next we identify those dimensions from ``I_\\text{neg}`` where ``X`` is also
+positive, using another support-function query in each dimension, and collect
+the dimensions in the index set ``I_\\text{mix}``.
+Let us call the remaining dimensions
+(``I_\\text{neg} \\setminus I_\\text{mix}``) ``I_\\text{nonpos}``.
+For each dimension in ``j ∈ I_\\text{mix}`` we will apply an intersection with
+axis-aligned polyhedra.
+In particular, we distinguish two cases using half-spaces ``x_j ≤ 0`` and
+``x_j ≥ 0``, and then compute all possible combinations to intersect, using one
+half-space per dimension ``j ∈ I_\\text{mix}``.
+
+Next we project the intersections in all dimensions from ``i ∈ I_\\text{mix}``
+such that we used the half-space ``x_i ≤ 0`` in their computation, and in all
+dimensions ``j ∈ I_\\text{nonpos}`` irrespective of the half-space used.
+
+Finally, we take the union of the resulting sets.
+"""
+function compute_union_of_projections!(r::Rectification{N}) where {N<:Real}
+    X = r.X
+    n = dim(X)
+
+    # construct a polyhedron corresponding to an index vector and a bit vector
+    function construct_constraints(indices, bits)
+        P = HPolyhedron{N}()
+        for (i, b) in enumerate(bits)
+            direction = Approximations.SingleEntryVector(indices[i], n,
+                                                         (b ? -one(N) : one(N)))
+            hs = HalfSpace(direction, zero(N))
+            addconstraint!(P, hs)
+        end
+        return P
+    end
+
+    # project negative dimensions to zero
+    function construct_projection(set, negative_dimensions, mixed_dimensions,
+                                  mixed_dimensions_enumeration)
+            v = ones(N, n)
+            # set negative indices to zero
+            v[negative_dimensions] .= zero(N)
+        for (i, d) in enumerate(mixed_dimensions)
+            if mixed_dimensions_enumeration[i]
+                # reset mixed index to one again because the set is nonnegative
+                v[d] = one(N)
+            end
+        end
+        return Diagonal(v) * set
+    end
+
+    # identify dimensions where set is negative and mixed (positive/negative)
+    negative_dimensions = Vector{Int}()
+    mixed_dimensions = Vector{Int}()
+    mixed_dimensions_enumeration = BitArray(undef, 0)
+    for i in 1:n
+        d = Approximations.SingleEntryVector(i, n, -one(N))
+        if ρ(d, X) > zero(N)
+            push!(negative_dimensions, i)
+            d = Approximations.SingleEntryVector(i, n, one(N))
+            if ρ(d, X) > zero(N)
+                push!(mixed_dimensions_enumeration, false)
+                push!(mixed_dimensions, i)
+            end
+        end
+    end
+
+    if isempty(mixed_dimensions)
+        # no mixed dimensions: no intersections needed
+        if isempty(negative_dimensions)
+            # no negative dimensions: set is unaffected by rectification
+            r.cache.set = X
```

Comment:
The set `X` may be unaffected by the rectification. So if we change for `convert` we have to wrap it around `UnionSetArray`..

Reply:
True, so I will not use `convert`.